test/units/module_utils/facts/system/test_date_time.py

```diff
@@ -0,0 +1,79 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import datetime
+import time
+
+from units.compat.mock import Mock, patch
+from .. base import BaseFactsTest
+
+from ansible.module_utils.facts.system.date_time import DateTimeFactCollector
+
+
+class MockDatetime(datetime.datetime):
+    @classmethod
+    def now(cls):
+        return cls(2020, 7, 8, 12, 34, 56, 123456)
+
+    @classmethod
+    def utcnow(cls):
+        return cls(2020, 7, 8, 16, 34, 56, 123456)
+
+
+class TestDateTimeFacts(BaseFactsTest):
+    __test__ = True
+    gather_subset = ['!all', 'date_time']
+    valid_subsets = ['date_time']
+    fact_namespace = 'ansible_date_time'
+    collector_class = DateTimeFactCollector
+
+    def _mock_module(self):
+        mock_module = Mock()
+        mock_module.params = {'gather_subset': self.gather_subset,
+                              'gather_timeout': 10,
+                              'filter': '*'}
+        return mock_module
+
+    def test_collect(self):
+        module = self._mock_module()
+        datetime.datetime = MockDatetime
+        fact_collector = self.collector_class()
+        facts_dict = fact_collector.collect(module=module)
+
+        self.assertIsInstance(facts_dict, dict)
+        self.assertEqual(facts_dict['date_time']['year'], '2020')
+        self.assertEqual(facts_dict['date_time']['month'], '07')
+        self.assertEqual(facts_dict['date_time']['weekday'], 'Wednesday')
+        self.assertEqual(facts_dict['date_time']['weekday_number'], '3')
+        self.assertEqual(facts_dict['date_time']['weeknumber'], '27')
+        self.assertEqual(facts_dict['date_time']['day'], '08')
+        self.assertEqual(facts_dict['date_time']['hour'], '12')
+        self.assertEqual(facts_dict['date_time']['minute'], '34')
+        self.assertEqual(facts_dict['date_time']['second'], '56')
+        self.assertEqual(facts_dict['date_time']['epoch'], '1594226096')
+        self.assertEqual(facts_dict['date_time']['date'], '2020-07-08')
+        self.assertEqual(facts_dict['date_time']['time'], '12:34:56')
+        self.assertEqual(facts_dict['date_time']['iso8601_micro'], '2020-07-08T16:34:56.123456Z')
+        self.assertEqual(facts_dict['date_time']['iso8601'], '2020-07-08T16:34:56Z')
+        self.assertEqual(facts_dict['date_time']['iso8601_basic'], '20200708T123456123456')
+        self.assertEqual(facts_dict['date_time']['iso8601_basic_short'], '20200708T123456')
+#        self.assertEqual(facts_dict['date_time']['tz'], 'EDT')
+#        self.assertEqual(facts_dict['date_time']['tz_dst'], 'EDT')
+#        self.assertEqual(facts_dict['date_time']['tz_offset'], '-0400')
```

Comment:
There shouldn't be this many assertions in one test. The best practice is to test one behavior per test (most of the time this means having one assert per test).